src/modeler/AutoRest.Swagger/Validation/TrackedResourceListByImmediateParent.cs

```diff
@@ -0,0 +1,133 @@
+﻿using AutoRest.Swagger.Validation.Core;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using AutoRest.Core.Logging;
+using AutoRest.Core.Properties;
+using AutoRest.Swagger.Model;
+using AutoRest.Swagger.Model.Utilities;
+using System.Text.RegularExpressions;
+
+namespace AutoRest.Swagger.Validation
+{
+    public class TrackedResourceListByImmediateParent : TypedRule<Dictionary<string, Dictionary<string, Operation>>>
+    {
+        /// <summary>
+        /// Id of the Rule.
+        /// </summary>
+        public override string Id => "M3010";
+
+        /// <summary>
+        /// Violation category of the Rule.
+        /// </summary>
+        public override ValidationCategory ValidationCategory => ValidationCategory.RPCViolation;
+
+        /// <summary>
+        /// The template message for this Rule. 
+        /// </summary>
+        /// <remarks>
+        /// This may contain placeholders '{0}' for parameterized messages.
+        /// </remarks>
+        public override string MessageTemplate => Resources.TrackedResourceListByImmediateParentOperationMissing;
+
+        /// <summary>
+        /// The severity of this message (ie, debug/info/warning/error/fatal, etc)
+        /// </summary>
+        public override Category Severity => Category.Warning;
+
+        private static Regex pathRegEx = new Regex("/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/[^/]+/[^/]+/{[^/]+}.*/(\\w+)$", RegexOptions.IgnoreCase);
+
+        private KeyValuePair<string, string> GetChildResourceName(string path, Dictionary<string, Dictionary<string, Operation>> paths, Dictionary<string, Schema> definitions)
+        {
+            Regex pathRegEx = new Regex("/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/[^/]+/[^/]+/{[^/]+}.*/(\\w+)/{[^/]+}$", RegexOptions.IgnoreCase);
+            Match match = pathRegEx.Match(path);
+            KeyValuePair<string, string> result = new KeyValuePair<string, string>();
+            if (match.Success)
+            {
+                string childresourcename = match.Groups[1].Value;
```

Comment:
Initially, I tried with named regex. unfortunately, that is handing non word characters also as opposed to \w. That was the reason I moved to index. Since there is only one extraction, I do not see the problem here